From 974e9b862e503c50501079e6586f81918e94a849 Mon Sep 17 00:00:00 2001
From: He Zhe <zhe.he@windriver.com>
Date: Fri, 19 Apr 2019 17:57:48 +0800
Subject: [PATCH] file01.sh: Fix in was not recognized

Some file has "pie" appending after LSB or MSB, which causes mismatch and the
following error.

"file01 10 TFAIL: in: was not recognized"
..."ELF 64-bit LSB pie executable"...

This patches tunes the regulation expression to include those cases.

Upstream-Status: Submitted [http://lists.linux.it/pipermail/ltp/2019-April/011758.html]

Signed-off-by: He Zhe <zhe.he@windriver.com>
---
 testcases/commands/file/file01.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/testcases/commands/file/file01.sh b/testcases/commands/file/file01.sh
index 0a8119e..55c0433 100755
--- a/testcases/commands/file/file01.sh
+++ b/testcases/commands/file/file01.sh
@@ -91,7 +91,9 @@ do_test()
 	 9) file_test in.m4 "M4 macro processor script, ASCII text" \
 			    "ASCII M4 macro language pre-processor text";;
 	10) file_test in "ELF .*-bit $TEST_ARCH executable, .*" \
-			 "ELF .*-bit $TEST_ARCH shared object, .*";;
+			 "ELF .*-bit $TEST_ARCH shared object, .*" \
+			 "ELF .*-bit $TEST_ARCH pie executable, .*" \
+			 "ELF .*-bit $TEST_ARCH pie shared object, .*";;
 	11) file_test in.ar "current ar archive";;
 	12) file_test in.tar "tar archive";;
 	13) file_test in.tar.gz "gzip compressed data, .*";;
-- 
2.7.4

